### PR TITLE
Improve Soroban reporting during upgrades and add a new HTTP endpoint

### PR DESCRIPTION
**Pull request by marta-lokhova**
Resolves https://github.com/stellar/stellar-core/issues/3960

This is useful for debugging/monitoring, but also in supercluster to quickly verify that upgrades went through.

### CODE REVIEW

File: src/main/CommandHandler.cpp [hunk 1]
```diff
@@ -711,6 +712,107 @@ CommandHandler::scpInfo(std::string const& params, std::string& retStr)
...
+        res["tx"]["max_size_bytes"] = conf.txMaxSizeBytes();
+        res["fee_transaction_size_1kb"] =
+            (Json::Int64)conf.feeTransactionSize1KB();
+
+        // General execution ledger settings
+        res["ledger"]["max_tx_count"] = conf.ledgerMaxTxCount();
+
+        // State archival settings
+        auto& archivalInfo = res["state_archival"];
+        auto stateArchivalSettings = conf.stateArchivalSettings();
```
Comment (SirTyson) [inline src/main/CommandHandler.cpp:781]: Nit: `auto const& stateArchivalSettings`

File: src/main/CommandHandler.cpp [hunk 2]
```diff
@@ -711,6 +712,107 @@ CommandHandler::scpInfo(std::string const& params, std::string& retStr)
...
+
+        // Contract data
+        res["max_contract_data_key_size"] = conf.maxContractDataKeySizeBytes();
+        res["max_contract_data_entry_size"] =
+            conf.maxContractDataEntrySizeBytes();
+
+        // Compute settings
+        res["tx"]["max_instructions"] = (Json::Int64)conf.txMaxInstructions();
+        res["ledger"]["max_instructions"] =
+            (Json::Int64)conf.ledgerMaxInstructions();
```
Comment (SirTyson) [inline src/main/CommandHandler.cpp:739]: Nit: `static_cast` here and below.

**Review by SirTyson — APPROVED** [review]
LGTM

File: src/main/CommandHandler.cpp (hunk 1 above)
Comment (marta-lokhova) [reply src/main/CommandHandler.cpp]: fixed

File: src/main/CommandHandler.cpp (hunk 2 above)
Comment (marta-lokhova) [reply src/main/CommandHandler.cpp]: updated

**Comment by sisuresh** [comment]
r+ f9a55e3af649f8ae3cbf672b125385a85d9d9d99

**Pull request merged**